### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by spuiuk**
Add the test frame work and tests in its own branch. This branch will be checked out and used on the client machines. 

The test framework relies on the test-info.yml file being created in the /root directory at the end of the execution of the ansible scripts from the master branch. 

Once the branch has been merged and is available in the gluster/samba-integration repo, I will add an additional PR to execute these tests.

### CODE REVIEW

**Comment by spuiuk** [comment]
At the moment moment, the smbtorture tests smb2.dir and smb2.read fail causing the tests to fail. I wonder if I should remove the tests for the time being while we figure out what is causing the issue.

**Comment by obnoxxx** [comment]
In order to make these PRs test-able, I suggest we could move forward the following way:

1. Merge minimal test infra and a test here in this repo that is trivial and passes.
2. Integrate the pulling of this test repo/branch in the master repo.
3. Establish PR checking from this PR in the centos-ci jobs.
4. Pursue this bigger PR with more tests running.

Makes sense?

**Comment by spuiuk** [comment]
> Makes sense?
Yep. Will push with mount test only for now.

File: test-info.yml.example [hunk 1]
```diff
@@ -0,0 +1,16 @@
+private_interfaces:
+  - "192.168.122.100"
+  - "192.168.122.101"
```
Comment (obnoxxx) [inline test-info.yml.example:3]: these addresses will need to be provided by the test setup

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,2 @@
+tests:
```
Comment (obnoxxx) [inline Makefile:1]: Should we use the (afaict) more standard `test` target?

```suggestion
test:
```

**Pull request merged**